src/core/shards/strings.cpp

```diff
@@ -187,7 +189,8 @@ struct Join {
     _buffer.append(input.payload.seqValue.elements[0].payload.stringValue, SHSTRLEN(input.payload.seqValue.elements[0]));
 
     for (uint32_t i = 1; i < input.payload.seqValue.len; i++) {
-      assert(input.payload.seqValue.elements[i].valueType == SHType::String);
+      assert(input.payload.seqValue.elements[i].valueType == SHType::String ||
+             input.payload.seqValue.elements[i].valueType == SHType::Bytes);
       _buffer.append(_separator);
       _buffer.append(input.payload.seqValue.elements[i].payload.stringValue, SHSTRLEN(input.payload.seqValue.elements[i]));
```

Comment:
Here we use `append()` while in `Bytes.Join` we use `insert()`. I assume they are equivalent but I think we should be consistent in both implementations and use only one way of doing it.